### PR TITLE
Automatic tests

### PR DESCRIPTION
**Pull request by orhoj**
## Purpose
Add initial test coverage. Note that for now we only test negative scenarios as we do not have a node or a mock server to test against right now.

## Changes
- Added automatic tests for negative validation scenarios.
- Refactored validation of expected parameters.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: src/server.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        nodePort,
+        insecureCredentials,
+        metadata,
+        nodeTimeout
+    );
+    const server = new jayson.Server(getJsonRpcMethods(nodeClient));
+
+    const app = express();
+    app.use(express.json());
+    app.post('/json-rpc', (req, res, next) => {
```
Comment (shjortConcordium) [inline src/server.ts:29]: remove json-rpc maybe :pleading_face:

File: test/methods.test.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+afterAll(() => {
+    jsonRpcServer.close();
+});
+
+test('missing params fails', async () => {
+    const addressRequestWithMissingParam = createRequest('getNextAccountNonce');
+    const response = await request(app)
+        .post('/json-rpc')
```
Comment (shjortConcordium) [inline test/methods.test.ts:23]: Remember to change these, if you change it in the code

File: src/server.ts (hunk 1 above)
Comment (shjortConcordium) [reply src/server.ts:29]: Or make it a parameter?

Comment (orhoj) [reply src/server.ts]: I have just removed it. It is in the interest of everyone for it to be somewhat consistent across people who run the server.

**Review by shjortConcordium — APPROVED** [review]

**Pull request merged**